controller/token.go

```diff
@@ -291,6 +292,48 @@ func (c *TokenController) Delete(ctx *app.DeleteTokenContext) error {
 	return ctx.OK([]byte{})
 }
 
+func (c *TokenController) Exchange(ctx *app.ExchangeTokenContext) error {
```

Comment:
The code here looks good, I have a question about the function name though - is "Exchange" significant in some way?  To me, it doesn't seem as intuitive as something like AuthenticateServiceAccount for example.

Reply:
I used OAuth 2 term "Exchange" for token exchange action because I was thinking of re-using this action to support other "grant_type" too. Right now we support "client_credentials" only. But latter we should support "authorization_code" too. See https://github.com/fabric8-services/fabric8-auth/issues/145

Reply:
No problem, perhaps this could be described in the function comment? E.g. for now we can have:

// This function provides token exchange, allowing clients to authenticate using an id and secret value
// May be expanded in the future to support other exchange types

Reply:
Yeah.. I will add the comment. Good idea.